# the-Needleman-Wunsch
Pairwise Sequence Global Alignment implementation

Implementing the Needleman-Wunsch global sequence alignment algorithm with both linear and affine gap penalties. This algorithm takes two amino acid sequences and gap penalty parameters as input, providing the maximum alignment score along with the alignment achieving this score. In cases of multiple optimal alignments, the algorithm outputs one of them.

Features:
Global Alignment: Utilizes the Needleman-Wunsch algorithm for global sequence alignment.
Linear and Affine Gap Penalties: Supports both linear and affine gap penalties for versatile alignment scenarios.
BLOSUM62 Scoring Matrix: Employs the BLOSUM62 scoring matrix for match and mismatch scoring.
Usage:
Input two amino acid sequences and gap penalty parameters.
Obtain the maximum alignment score and the corresponding aligned sequences.


Assignment 1 of CMPE549 - Bioinformatics course at Bogazici University.
